src/content-helper/editor-sidebar/smart-linking/component.tsx

```diff
@@ -139,27 +134,75 @@ export const SmartLinkingPanel = ( {
 			applyTo: getApplyTo(),
 			retrying: isRetrying(),
 			retryAttempt: getRetryAttempt(),
+			smartLinks: getSmartLinks(),
 		};
 	}, [] );
 
+	/**
+	 * The filtered list of smart links that have been applied.
+	 *
+	 * This list is memoized to prevent unnecessary re-renders.
+	 *
+	 * @since 3.16.0
+	 */
+	const appliedLinks = useMemo( () => smartLinks.filter( ( link ) => link.applied ), [ smartLinks ] );
+
 	/**
 	 * Loads the Smart Linking store actions.
 	 *
 	 * @since 3.14.0
 	 */
 	const {
+		setIsReady,
 		setLoading,
 		setError,
-		setSuggestedLinks,
+		addSmartLinks,
 		addOverlayBlock,
 		removeOverlayBlock,
 		setSmartLinkingSettings,
 		setApplyTo,
 		setMaxLinks,
 		setIsRetrying,
 		incrementRetryAttempt,
+		purgeSmartLinksSuggestions,
+		setIsReviewModalOpen,
 	} = useDispatch( SmartLinkingStore );
 
+	/**
+	 * Handles the initialization of the Smart Linking existing links.
+	 *
+	 * @since 3.16.0
+	 */
+	useEffect( () => {
+		if ( ! ready ) {
+			const existingSmartLinks = getAllSmartLinksInPost();
+			addSmartLinks( existingSmartLinks ).then( () => {
+				setIsReady( true );
+			} );
+		}
+	}, [ addSmartLinks, ready, setIsReady ] );
+
+	/**
+	 * Handles the ending of the review process.
+	 * Shows a success notice if the review is done and there are added links.
+	 *
+	 * @since 3.16.0
+	 */
+	useEffect( () => {
+		if ( ! isReviewDone ) {
+			setNumAddedLinks( 0 );
+		} else if ( numAddedLinks > 0 ) {
+			createNotice(
+				'success',
+				/* translators: %d: number of smart links applied */
+				sprintf( __( '%s smart links successfully applied.', 'wp-parsely' ), numAddedLinks ),
+				{
+					type: 'snackbar',
+				},
+			);
+		}
+	}, [ isReviewDone ] ); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
This notice should only be shown when the the review is marked as done. Since numAddedLinks is updated as soon as a new link is applied, we don't want it as a dependency. Makes sense?